### PR TITLE
ENGDOCS-1944

### PR DESCRIPTION
**Pull request by aevesdocker**
Follow on from moving the Compose Spec into docs. 

This PR:
- Improves the initial TOC flow in the Compose manuals section.
    -  We now have an 'intro bit' which contains content that isn't heavily looked at and makes the toc appear shorter 
- Adapts some Compose spec content (regarding the compose file and compose application model) into a broader 'How Compose works' page, to give novice users a better understanding of how the compose file and the Spec and the CLI all fit together 
- Fixes the 'key features' page into a 'why use compose' with broader, more punchy benefits of Compose. 
- Converts the mermaid diagram into a docs-approved diagram with Docker colors. 

TL;DR = Not a lot of new content, just tweaking of existing. 

Planning on merging this asap

### CODE REVIEW

File: content/compose/_index.md [hunk 1]
```diff
@@ -42,10 +50,11 @@ aliases:
...
 {{< include "compose-eol.md" >}}
 
-Compose is a tool for defining and running multi-container Docker applications.
-With Compose, you use a YAML file to configure your application's services.
-Then, with a single command, you create and start all the services
-from your configuration.
+Docker Compose is a tool for defining and running multi-container applications. 
+It is the key to unlocking a streamlined and efficient development and deployment experience. 
+
+Compose simplifies the orchestration of your entire application stack, making it easy to manage services, networks, and volumes in a single, comprehensible YAML configuration file. Then, with a single command, you create and start all the services
```
[the remark this answers is not in the document]
Comment (ndeloof) [reply content/compose/_index.md:56]: we use to say compose is an "automation" to make it more explicit it runs adequate commands in sequence, but doesn't support any kube-style long-running "orchestration"

File: content/compose/compose-application-model.md [hunk 2]
```diff
@@ -23,16 +28,34 @@ A [secret](compose-file/09-secrets.md) is a specific flavor of configuration dat
...
+## The Compose file
 
-Project names must contain only lowercase letters, decimal digits, dashes, and underscores, and must begin with a lowercase letter or decimal digit.
+The default path for a Compose file is `compose.yaml` (preferred) or `compose.yml` that is placed in the working directory.
+Compose also supports `docker-compose.yaml` and `docker-compose.yml` for backwards compatibility of earlier versions.
+If both files exist, Compose prefers the canonical `compose.yaml`.
+
+You can use [fragments](compose-file/10-fragments.md) and [extensions](compose-file/11-extension.md) to keep your Compose file efficient and easy to maintain.
+
+Multiple Compose files can be [merged](13-merge.md) together to define the application model. The combination of YAML files are implemented by appending or overriding YAML elements based on the Compose file order you set.
```
Comment (stephaurelio) [inline content/compose/compose-application-model.md:48]: It looks like this verb should agree with the singular `combination`: 
```suggestion
Multiple Compose files can be [merged](13-merge.md) together to define the application model. The combination of YAML files is implemented by appending or overriding YAML elements based on the Compose file order you set. 
```

File: content/compose/compose-application-model.md [hunk 3]
```diff
@@ -23,16 +28,34 @@ A [secret](compose-file/09-secrets.md) is a specific flavor of configuration dat
...
+
+You can use [fragments](compose-file/10-fragments.md) and [extensions](compose-file/11-extension.md) to keep your Compose file efficient and easy to maintain.
+
+Multiple Compose files can be [merged](13-merge.md) together to define the application model. The combination of YAML files are implemented by appending or overriding YAML elements based on the Compose file order you set. 
+Simple attributes and maps get overridden by the highest order Compose file, lists get merged by appending. Relative
+paths are resolved based on the first Compose file's parent folder, whenever complimentary files being
+merged are hosted in other folders. As some Compose file elements can both be expressed as single strings or complex objects, merges apply to
+the expanded form. For more information, see [Working with multiple Compose files](multiple-compose-files/_index.md)
+
+If you want to reuse other Compose files, or factor out parts of you application model into separate Compose files, you can also use [`include`](compose-file/14-include.md). This is useful if your Compose application is dependent on another application which is managed by a different team, or needs to be shared with others.
```
Comment (stephaurelio) [inline content/compose/compose-application-model.md:54]: Small typo: 
```suggestion
If you want to reuse other Compose files, or factor out parts of your application model into separate Compose files, you can also use [`include`](compose-file/14-include.md). This is useful if your Compose application is dependent on another application which is managed by a different team, or needs to be shared with others.
```

File: content/compose/compose-application-model.md [hunk 4]
```diff
@@ -42,40 +65,7 @@ The backend stores data in a persistent volume.
...
-        style id2 stroke:#000,stroke-width:1px,stroke-dasharray: 10
-        B2 ==r+w==> id3[(Persistent\nVolume)]
-    end
-    style A fill:#eeeeee, stroke-width:0px
-    direction LR
-    id4[External\nUser] ---id5(((443)))--->|Frontend\nNetwork| B1
-    style id4 stroke:#000,stroke-width:2px
-    B1 --Backend\nNetwork--> B2
-```
+![Compose application example](images/compose-application.webp)
```
Comment (stephaurelio) [inline content/compose/compose-application-model.md:68]: Looks great! I actually couldn't see the mermaid one in either Chrome or Safari in the current iteration of this page so this file format works way better.

File: content/compose/compose-application-model.md [hunk 5]
```diff
@@ -126,8 +116,8 @@ networks:
   back-tier: {}
 ```
 
-This example illustrates the distinction between volumes, configs and secrets. While all of them are all exposed
-to service containers as mounted files or directories, only a volume can be configured for read+write access.
-Secrets and configs are read-only. The volume configuration allows you to select a volume driver and pass driver options
-to tweak volume management according to the actual infrastructure. Configs and secrets rely on platform services,
-and are declared `external` as they are not managed as part of the application lifecycle. Compose uses a platform-specific lookup mechanism to retrieve runtime values.
\ No newline at end of file
+## What's next
```
Comment (stephaurelio) [inline content/compose/compose-application-model.md:119]: Super helpful to have this!

File: content/compose/intro/features-uses.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+---
+description: Key benefits and use cases of Docker Compose
```
Comment (stephaurelio) [inline content/compose/intro/features-uses.md:2]: Really liked this page as a starting point, especially the What's next links at the end 👍

File: content/compose/reference/_index.md [hunk 7]
```diff
@@ -176,6 +176,25 @@ underscores, and must begin with a lowercase letter or decimal digit. If the
 `basename` of the project directory or current directory violates this
 constraint, you must use one of the other mechanisms.
 
+### Have multiple isolated environments on a single host
```
[the remark this answers is not in the document]
Comment (stephaurelio) [reply content/compose/reference/_index.md:179]: It took me a minute to find where this page is in the docs! Is there a direct link to this reference from the manuals?

**Review by stephaurelio — COMMENTED** [review]
The new ToC looks great! It's much more readable and gives a better starting place for being new to Compose than the current ToC. What's next links are super helpful, and I found the How Compose works page to be clearer than what we have now, plus avoids less navigation to get the information. Left a few small comments and impressions ✨

**Review by stephaurelio — APPROVED** [review]
LGTM!
